tests/index.js

```diff
@@ -23,6 +23,20 @@ describe('all rule files should be exported by the plugin', function() {
   });
 });
 
+describe('deprecated rules', function() {
+  it('marks all deprecated rules as deprecated', function() {
```

Comment:
lol i see this in your OP now. yeah i think it might be worth it - `getAllRules().filter(x => x.deprecated)` seems like it should be a viable approach